master/internal/api_trials.go

```diff
@@ -53,6 +54,51 @@ var (
 	TrialAvailableSeriesBatchWaitTime = 15 * time.Second
 )
 
+func (a *apiServer) timeToFloat(t time.Time) float64 {
+	// If time.Time is the empty value, UnixNano will return the farthest back
+	// timestamp a float can represent, which is some large negative value.
+	if t.IsZero() {
+		return 0
+	}
+	return float64(t.UnixNano()) / 1e9
+}
+
+func (a *apiServer) scanMetricsSeries(rows *sql.Rows) (
+	metricSeriesBatch, metricSeriesTime, metricSeriesEpoch []lttb.Point, maxEndtime time.Time,
+) {
+	var maxEndTime time.Time
+	var avgMetrics map[string]float64
+	for rows.Next() {
+		var batches uint
+		var value float64
+		var endTime time.Time
+		var metrics *string
+		err := rows.Scan(&batches, &value, &endTime, &metrics)
+		if err != nil {
+			continue
+		}
+		if metrics != nil {
+			err = json.Unmarshal([]byte(*metrics), &avgMetrics)
+			if err != nil {
+				continue
+			}
+		}
+		metricSeriesBatch = append(metricSeriesBatch, lttb.Point{X: float64(batches), Y: value})
+		metricSeriesTime = append(metricSeriesTime, lttb.Point{X: a.timeToFloat(endTime), Y: value})
+		// For now we will always search for an "epoch" value but this can be updated in the future
+		// to accept or expect a dynamic list of poossible x-axis values.
+		epoch, ok := avgMetrics["epoch"]
```

Comment:
not a big deal, but is there any way we can pass this hardcoded key names in to this method? i imagine in the future we want to support more generic default x-axis, so this list of hard-coded key names might get obscured in this function.

Reply:
upon further investigation and refactoring I actually would rather hold off on this. Only because without implementing more logic around truly handling custom metrics not much else would be changed, i.e we'd still just be special casing logic around looking for `epoch` and appending data to the `metricSeriesEpoch` array and we would not gain much for now.

Reply:
@azhou-determined I did make this update after all https://github.com/determined-ai/determined/pull/5946/files#diff-27e30348f214ed09d8fa135cbbc1ff709dc67e9e3cc378801614e956828d7d86R572

Reply:
Also added a note to the PR description about the `xAxisMetricLabels `